### DISCUSSION TITLE
What cool things have you built with DG?

### DISCUSSION THREAD

**Discussion by BekahHW** [deepgram/community#21] [General help]
## We want to see your projects :eyes:

Have you built something cool with Deepgram? We want to see and share it. Drop a link below with a short description, and let us know what inspired you.

**Comment by BekahHW** [comment]
I created a React project that uses Deepgram to transcribe my daily affirmations! You can check it out here: [blog post](https://blog.deepgram.com/how-to-add-speech-recognition-to-your-react-project/).

**Reply by madzadev** [reply]
Such a cool project @BekahHW and well explained 👍✨💯

**Reply by BekahHW** [reply]
Thanks, @madzadev! You should put your project in this thread so others can find it!

**Comment by Cerchie** [comment]
I converted the Deepgram API to GraphQL and then consumed it from a Next.js frontend for a [how-to blog post](https://stepzen.com/blog/consuming-the-deepgram-api-the-graphql-edit)! (by the way, I don't work at StepZen anymore)

**Reply by michaeljolley** [reply]
That is nice! I had someone ask me about a GraphQL endpoint to the API this past week! I may point them to your post.

**Comment by MooseSaeed** [comment]
Deepgram inspired me to create cool projects using it. I was particularly inspired by how clear the documentation is.

Here are my projects:

## 1- Wordleit (Deepgram Hackathon Grand Prize Winner Project) :
Github: https://github.com/MooseSaeed/wordleup

![Image description](https://dev-to-uploads.s3.amazonaws.com/uploads/articles/ejr7d77zv0jhg3epbfxv.png)

- **Wordleit Editor** is an open source markdown text editor that gives you a seamless experience as both a reader and a writer. Supported with AI Speech Recognition. Text to speech - Speech to text - Mp3 to Text.

Articles:
[Introduction](https://dev.to/moose_said/write-articles-with-ai-speech-recognition-and-live-preview-508k) - [Vocal Commands](https://dev.to/moose_said/use-vocal-commands-to-apply-markdown-to-your-article-99) - [TTS and STT Feature](https://dev.to/moose_said/text-editor-with-text-to-speech-and-speech-to-text-4l37)


- **Wordleit Streamer** is an open source video streamer and online transcriptor tool.

Article:
[Overview](https://dev.to/moose_said/online-video-streamer-and-audio-transcription-2ein)

- **Wordleit Grammy** is an AI speaking Bot created to serve your purposes whatever it is.

Article:
[Overview](https://dev.to/moose_said/add-ai-robot-to-your-portfolio-or-company-website-3045)

## 2- BeReadable:
Github: https://github.com/MooseSaeed/be-readable

![BeReadable Logo Animation](https://dev-to-uploads.s3.amazonaws.com/uploads/articles/6nfwwx0650bqkyn6r3ni.gif)

**BeReadable** is a simple online tool that helps the user to turn audio into words and it can also record the audio and the user can play it or download it to mp3. BeReadable supports a lot of languages as it sends the user's language selection to Deepgram API and return back the correct transcription for the selected language. The tool is a straight to the point kind of tool, it doesn't have any complexity in using and the UI is quite simple. It has two main features and both of them are incredibly important.

Article:
[Overview](https://dev.to/moose_said/bereadable-online-multilingual-audio-transcription-and-recorder-a8a)

## 3- Tailwinder (Appwrite Hackathon Grand Prize Winner Project):
Girhub: https://github.com/MooseSaeed/Tailwinder

![Tailwinder Logo](https://dev-to-uploads.s3.amazonaws.com/uploads/articles/3suhrjaxsp9jxo4bxa2w.gif)

**Tailwinder** is an open source community for tailwind fans. Deepgram was utilized as a substitute for writing a blog post with your voice. I had planned to improve it or include Wordleit Editor into it, but I never had the chance.

Article:
[Overview](https://dev.to/moose_said/introducing-tailwinder-home-for-tailwind-fans-powered-with-appwrite-4c0o)

**Reply by BekahHW** [reply]
So many great projects! Thanks for sharing.

**Reply by DuyguA** [reply]
They all look great!

**Comment by JesseRWeigel** [comment]
I used the web api to make a note taking app with React and Typescript. Right now a user pastes in their api key and can see what they say on the screen. I want to add the ability to save, edit, and delete notes. Just a little proof of concept to get me familiar with Deepgram. https://jesserweigel.github.io/voice-to-text-notes/

**Reply by JesseRWeigel** [reply]
I am going to be updating this project throughout October for Hacktoberfest! Will let you know if it ends up being awesome.

**Reply by BekahHW** [reply]
I'll add it to our Hacktoberfest blog post next week!

**Comment by Dhravya** [comment]
I created a python script that automatically renders subtitles into a video, using deepgram! [I call it DeepSubtitles](https://github.com/dhravya/deepsubtitles)

I haven't worked on the project a lot after the hackathon, and can probably improve a lot, maybe make a webapp or just an API that easily does it!

The subtitles in the GIF below are generated by deepgram and baked into the video using deepsubtitles

There are a lot more use cases for this too, maybe I can add "Themes" so people can use their own fonts and sizes etc.
![Small GIF of how it works](https://camo.githubusercontent.com/558cbbbda72f136829977bc9aac096d7d6b53f85d3a5d6d9480b530c0bb943b0/68747470733a2f2f6a2e676966732e636f6d2f3537775138762e676966)

**Reply by BekahHW** [reply]
I love this project! If you have ideas for how to add to it, you might want to opt your repository into Hacktoberfest so folks can contribute to your project. If you're interested, I'm happy to help you get started.

**Reply by Dhravya** [reply]
Yes I'll add the repository to Hacktoberfest!

**Reply by michaeljolley** [reply]
That is very cool @Dhravya!

**Comment by zainbinfurqan** [comment]
I Created the Tic Tac Toe game in React.js using voice to text feature of Deepgram.
This was my first time to participate in this type of hackathon & I must say I learn a lot from this project .

- Checkout the article here : [lets-play-voice-tic-tac-toe](https://dev.to/zainbinfurqan/lets-play-voice-tic-tac-toe-43fi)
- Here is my YouTube video about my experience as participant  in Hackathon : [Deepgram X Dev.to](https://www.youtube.com/watch?v=UcQVOob_oo8)

**Reply by BekahHW** [reply]
I love seeing the fun projects we can learn from. Thanks for posting 🎉

**Comment by shirgoldbird** [comment]
I built a [Discord bot](https://github.com/shirgoldbird/deepgram-discord/) that transcribes Discord voice calls into a text channel! It uses webhooks to make the transcriptions look like they're coming from the user who's currently speaking.

**Reply by BekahHW** [reply]
That's super cool, Shir! I can't wait to check this out.

**Comment by maboa** [comment]
I built an open source [transcript editor](https://github.com/hyperaudio/hyperaudio-lite-editor) that preserves word timings and allows you to export interactive transcripts and captions.

[A demo can be found here.](https://lab.hyperaud.io/demos/hle/demo-05/deepgram.html)

**Reply by BekahHW** [reply]
This is really awesome! What are your plans for the project?

**Reply by michaeljolley** [reply]
Oh, I love this!

**Reply by maboa** [reply]
Thanks so much for the support! I have found Deepgram really easy to use and having used many similar APIs have to say the quality, punctuation and speed (especially the speed) of transcription are second to none (at least in English and Italian)! 

The idea is to continue to add features, such as the ability to correct captions and perhaps even translate them into various languages.

I'll probably add the various models too, such as phone voice etc, and maybe also add the option to auto-detect language as default.

Regarding the above – It would be great to be able to query the API to understand which models are supported by which languages, otherwise I can take two approaches:

1. Check to see if the request fails with certain parameters and retry changing the parameters if it does. (I do this already with the enhanced models).
2. Manually keep a look-up-table of the model support for each language.

I have an Electron build on a separate branch, but it seems a bit heavy considering how light the actual JavaScript is, so I'd like to make a smaller standalone version using [Tauri](https://tauri.app/) or just create a Progressive Web App that works in the browser, but can also be installed as standalone.

I'd also like to integrate this into our [Hyperaudio Wordpress Plugin](https://github.com/hyperaudio/wordpress-hyperaudio) and am considering adding to our Hyperaudio for Conferences codebase (in place of the Amazon transcription API) of which you can see an example here: [mozfest.hyper.audio](https://mozfest.hyper.audio/)

**Reply by geekchick** [reply]
This is amazing! Great work @maboa !

**Reply by maboa** [reply]
Just a heads up that we've published a new version with localStorage support https://lab.hyperaud.io/demos/hle/demo-06/deepgram.html

**Reply by maboa** [reply]
And an alternative version, with a different layout and installable as a Progressive Web App (on Chrome) https://lab.hyperaud.io/demos/hle/demo-09/alternative.html

**Reply by BekahHW** [reply]
Awesome!

**Reply by maboa** [reply]
Good news! - We're now using Deepgram's diarization (speaker differentiation) – you can check it out here https://hyperaudio.github.io/hyperaudio-lite-editor/alternative.html 

You can add speakers anywhere in the text between square brackets. They will appear as bold. Initially the speakers are labelled `[speaker-0]`, `[speaker-1]` etc 

<img width="711" alt="Screenshot 2023-02-14 at 10 53 07" src="https://user-images.githubusercontent.com/208756/218715057-c74daa43-e1ae-4ee8-9b0e-a7ad3b7d618f.png">

**Reply by BekahHW** [reply]
That's awesome! I just tested it out, and it looks good.

**Reply by maboa** [reply]
Thanks @BekahHW !

<img width="548" alt="Screenshot 2023-02-15 at 11 20 22" src="https://user-images.githubusercontent.com/208756/219013716-0aad9452-e66e-4f68-97ac-9eb85487757a.png">

Although I am seeing a couple of what appears to be "off-by-one" errors (see [speaker-1], [speaker-3] and [speaker-4]) in the above transcript. Perhaps this occurs because the diarization and transcription processes are separate, and so timings differ slightly. 

It should be :

[speaker-1] We are, as people...
[speaker-3] I think that unfortunately...
[speaker-4] When is healthcare...

I did initially try to shift the speaker titles by one word (looking at the punctuation), but the code became a little complex, and we probably need to account for the situation where speakers interrupt each other, which is difficult to distinguish.

Unrelated – I'm not sure what the capitalization signifies in paragraph 4 – an increase in volume perhaps? This seem to happen when  transcribing using English US. English GB does not produce captilized text (in this case).

[EDIT] In case it's useful the file I'm transcribing is https://lab.hyperaud.io/video/BBC/education.mp4

**Comment by Sumanth077** [comment]
I have created a Steamship Plugin for Deepgram AI, which makes Transcription and summarization of any Video/Audio easy with a couple of clicks. 🚀

Checkout the Plugin and also the Readme file on how easily you can use this plugin in your next Project. 

https://github.com/Sumanth077/Deepgram-AI-Plugin 🔗 cc: @BekahHW

**Reply by BekahHW** [reply]
Awesome, @Sumanth077!
